### PR TITLE
Fix cloning of check-in status in registration forms

### LINKED ISSUE #4208: When cloning registration forms, check-in status gets cloned too
**Describe the bug**
The states of participants seem to be cloned along with the participants themselves.

**To Reproduce**
1. Create an event
2. Add some participants and set their check-in status
3. Clone the event

**Expected behavior**
The check-in state should have been reset.

### PR DESCRIPTION
**Pull request by Leats**
Closes #4208.

### CODE REVIEW

File: indico/modules/events/registration/clone_test.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        checked_in=True,
+        state=1,
+        currency="USD",
+        email="1337@example.com",
+        user=dummy_user
+    )
+    db.session.flush()
+
+    assert dummy_user.registrations.filter_by(event=dummy_event).first().full_name == 'Guinea Pig'
+    assert dummy_event.registration_forms[0].active_registrations[0].user == dummy_user
```
Comment (ThiefMaster) [inline indico/modules/events/registration/clone_test.py]: `dummy_event.registration_forms[0]` is already available in `dummy_regform`

File: indico/modules/events/registration/clone_test.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    assert dummy_event.registration_forms[0].active_registrations[0].checked_in is True
```
Comment (ThiefMaster) [inline indico/modules/events/registration/clone_test.py]: omit the `is True`

File: indico/modules/events/registration/clone_test.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    copied_event = create_event()
+    EventCloner.run_cloners(dummy_event, copied_event, {'registrations', 'registration_forms'})
+
+    assert dummy_user.registrations.filter_by(event=copied_event).first().full_name == 'Guinea Pig'
+    assert copied_event.registration_forms[0].active_registrations[0].user == dummy_user
+    assert copied_event.registration_forms[0].active_registrations[0].checked_in is False
```
Comment (ThiefMaster) [inline indico/modules/events/registration/clone_test.py]: `assert not ...`

Comment (ThiefMaster) [reply indico/modules/events/registration/clone_test.py]: I'd assign `copied_event.registration_forms[0].active_registrations[0]` to `copied_reg` or something like that

File: indico/modules/events/registration/clone_test.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+from indico.core.db import db
+from indico.modules.events.cloning import EventCloner
+from indico.modules.events.features.util import set_feature_enabled
+from indico.modules.events.registration.models.registrations import Registration, RegistrationState
+
+
+pytest_plugins = 'indico.modules.events.registration.testing.fixtures'
+
+
+def test_registration_clone(dummy_event, dummy_regform, create_event, dummy_user):
```
Comment (pferreir) [inline indico/modules/events/registration/clone_test.py]: So nice to see people sending PRs with tests!

**Pull request merged**